### PR TITLE
CB-15919 Eliminate N+1 queries during the autosyncing initalization. …

### PR DESCRIPTION
**Pull request by topolyai5**
…We are fetching the pure entity but we only use the crn and id. Because of lazy/eager fetch, and toString combination we can fetch unnecesary relations. I have refactored in all services

See detailed description in the commit message.

### CODE REVIEW

File: common/src/main/java/com/sequenceiq/cloudbreak/quartz/statuschecker/service/StatusCheckerJobService.java [hunk 1]
```diff
@@ -76,13 +79,25 @@ public void schedule(Long id, Class<? extends JobResourceAdapter<?>> resource) {
...
+        schedule(jobDetail, trigger, resource.getJobResource());
+    }
+
+    public <T> void scheduleLongIntervalCheck(Long id, Class<? extends JobResourceAdapter<?>> resourceAdapterClass) {
+        try {
+            Constructor<? extends JobResourceAdapter> c = resourceAdapterClass.getConstructor(Long.class, ApplicationContext.class);
+            JobResourceAdapter resourceAdapter = c.newInstance(id, applicationContext);
+            scheduleLongIntervalCheck(resourceAdapter);
+        } catch (NoSuchMethodException | IllegalAccessException | InstantiationException | InvocationTargetException e) {
+            LOGGER.error(String.format("Error during scheduling quartz job: %s", id), e);
```
Comment (foldik) [inline common/src/main/java/com/sequenceiq/cloudbreak/quartz/statuschecker/service/StatusCheckerJobService.java:93]: No need to do `String.format`. Can you please change it to:

```java
LOGGER.error("Error during scheduling quartz job: {}", id, e);
```

The exception will be logged at the and.

**Review by foldik — APPROVED** [review]
LGTM

File: core/src/main/java/com/sequenceiq/cloudbreak/job/AbstractStackJobInitializer.java [hunk 2]
```diff
@@ -1,34 +1,19 @@
...
 
-    protected Stream<Stack> getAliveStacksStream() {
-        return stackService.getAllAlive().stream()
-                .map(this::convertToStack);
-    }
-
-    protected Stream<Stack> getAliveAndNotDeleteInProgressStacksStream() {
-        return getAliveStacksStream()
-                .filter(s -> !s.isStackInDeletionOrFailedPhase());
-    }
```
Comment (nadamdb) [inline core/src/main/java/com/sequenceiq/cloudbreak/job/AbstractStackJobInitializer.java:25]: As discussed these should not be deleted completely as there are initializers (e.g. StructuredSynchronizerJobInitializer) that should init jobs for every cluster besides delete_completed status ones and not just alive ones.

**Review by nadamdb — APPROVED** [review]
LGTM

**Pull request merged**